queue/retry.go

```diff
@@ -151,49 +200,44 @@ func (rh *BasicRetryHandler) waitForJob(ctx context.Context) {
 		rh.wg.Done()
 	}()
 
-	timer := time.NewTimer(0)
-	defer timer.Stop()
 	for {
 		select {
 		case <-ctx.Done():
 			return
-		case <-timer.C:
-			var j amboy.Job
+		case j := <-rh.pending:
+			rh.mu.Lock()
+			rh.tryReduceOverflow()
```

Comment:
As soon as one job is removed from the channel, another one has to be added if possible so that other threads can handle jobs ASAP. Otherwise, it would not replace the job that was just removed, and other threads might believe there's no work to do when there actually still is.

Reply:
Done.